app/uk/gov/hmrc/transitmovementsrouter/connectors/PushNotificationsConnector.scala

```diff
@@ -0,0 +1,91 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.connectors
+
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import cats.data.EitherT
+import com.google.inject.ImplementedBy
+import com.google.inject.Inject
+import com.google.inject.Singleton
+import io.lemonlabs.uri.UrlPath
+import play.api.http.HeaderNames
+import play.api.http.MimeTypes
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import play.api.http.Status.NOT_FOUND
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.http.StringContextOps
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.transitmovementsrouter.config.AppConfig
+import uk.gov.hmrc.transitmovementsrouter.models.MessageId
+import uk.gov.hmrc.transitmovementsrouter.models.MovementId
+import uk.gov.hmrc.transitmovementsrouter.models.errors.PushNotificationError
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.transitmovementsrouter.models.errors.PushNotificationError.MovementNotFound
+import uk.gov.hmrc.transitmovementsrouter.models.errors.PushNotificationError.Unexpected
+
+@ImplementedBy(classOf[PushNotificationsConnectorImpl])
+trait PushNotificationsConnector {
+
+  def post(movementId: MovementId, messageId: MessageId, source: Source[ByteString, _])(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): EitherT[Future, PushNotificationError, Unit]
+}
+
+@Singleton
+class PushNotificationsConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig: AppConfig) extends PushNotificationsConnector {
+
+  val baseUrl           = appConfig.transitMovementsPushNotificationsUrl
+  val baseRoute: String = "/transit-movements-push-notifications"
+
+  private def pushNotificationMessageUpdate(movementId: MovementId, messageId: MessageId): UrlPath =
+    UrlPath.parse(s"$baseRoute/traders/movements/${movementId.value}/messages/${messageId.value}")
+
+  override def post(movementId: MovementId, messageId: MessageId, source: Source[ByteString, _])(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): EitherT[Future, PushNotificationError, Unit] =
+    EitherT {
+      val url = baseUrl.withPath(pushNotificationMessageUpdate(movementId, messageId))
+
+      httpClientV2
+        .post(url"$url")
+        .addHeaders(HeaderNames.CONTENT_TYPE -> MimeTypes.XML)
+        .withBody(source)
+        .execute[Either[UpstreamErrorResponse, HttpResponse]]
```

Comment:
I think we can move this part to BaseConnector ? Just like below
    def executeAndExpect(expected: Int)(implicit ec: ExecutionContext): Future[Unit] =
      requestBuilder
        .execute[HttpResponse]
        .flatMap {
          response =>
            response.status match {
              case `expected` => Future.successful(())
              case _          => response.error
            }
        }

Reply:
This function returns an Either, if I was to use executeAndExpect which returns Future[Unit] I would need to still match on the error so I'm not sure this would make any improvement

Reply:
and since we are not making use of a service here and just calling the connector from the controller, we need to return an eitherT